### PR TITLE
Frontend - Maintenance Issue Page

### LINKED ISSUE #138: Create  GET endpoints for retrieving all Maintenance Issues and retrieving a Maintenance Issue by Id
**Is your feature request related to a problem? Please describe.**

Currently, there is no implementation on logging maintenance issues for the platform.

**Describe the solution you'd like**

As part of implementing the logging maintenance issue feature, when the GET endpoints are called, it will return the necessary Maintenance Issue(s).

### Checklist:

- [x] I have double checked that the issue does not currently exist. 
- [x] I have added the appropriate labels (e.g. feature, front-end, back-end).
- [x] I have added the **Platform Project** project

### LINKED ISSUE #139: Create View Maintenance Logs Page
**Is your feature request related to a problem? Please describe.**

A view logs page needs to be created for one of the maintenance logging features. This page will allow the user to view a specific issue, similar to the way users can currently view a specific bill card in more detail. 

**Describe the solution you'd like**

When the user clicks on a maintenance issue card, it will redirect them to a page that shows further details of the specific issue. E.g. the description of the issue that occurred, the date it was issued, the user who issued the card, and the issue resolve status.  

### Checklist:

- [x] I have double checked that the issue does not currently exist. 
- [x] I have added the appropriate labels (e.g. feature, front-end, back-end).
- [x] I have added the **Platform Project** project

### LINKED ISSUE #140: Create POST Maintenance Issue Endpoint
**Is your feature request related to a problem? Please describe.**

Need to add a POST endpoint that will enable users to post their maintenance issues to MongoDB database. 

**Describe the solution you'd like**

The POST maintenance endpoint will retrieve a Maintenance Issue object from the client and add it to the issues database. 

### Checklist:

- [x] I have double checked that the issue does not currently exist. 
- [x] I have added the appropriate labels (e.g. feature, front-end, back-end).
- [x] I have added the **Platform Project** project

### LINKED ISSUE #141: Frontend - Maintenance Issue Page
### Is your feature request related to a problem? Please describe.

This page allows users to log maintenance issues. Note that it does Not send these to an external party, it simply allows flatmates to keep track of who has sent maintenance requests to the landlord or relevant parties. A use example could be the window is broken, Flatmate A checks Flatshare and sees that Flatmate B has left an entry documenting them contacting the landlord. So Flatmate A knows they don't have to contact the landlord.

This feature helps towards the goal of "Maintenance/Repairs - Documentation of complaints/issues and similar stuff" from the project proposal.

### Describe the solution you'd like

Build a page with functionality similar to the bills page. Users should be able to create a new issue, resolve issues and view all past issues.

The frontpage design may also need to be adjusted to account for a link to this page. (Possibly axing the chores button? To be discussed)

### Describe alternatives you've considered

We have considered functionality where an email is sent to the landlord from the app, but decided it was out of scope for an MVP for this feature. 

### Checklist:

- [x] I have double checked that the issue does not currently exist. 
- [x] I have added the appropriate labels (e.g. feature, front-end, back-end).
- [x] I have added the **Platform Project** project

### LINKED ISSUE #142: Create Maintenance Issue Component
**Is your feature request related to a problem? Please describe.**

A component is needed to create a new issue. 

**Describe the solution you'd like**

This should be similar to the create bill component, replacing the fields and functionality as needed to match a maintenance issue.

**Additional context**

This will assist the completion of #141 

### Checklist:

- [x] I have double checked that the issue does not currently exist. 
- [x] I have added the appropriate labels (e.g. feature, front-end, back-end).
- [x] I have added the **Platform Project** project

### LINKED ISSUE #148: Create PUT issue endpoint
**Is your feature request related to a problem? Please describe.**
The current backend does not support updating an issue information such as adding extra details and changing resolved status

**Describe the solution you'd like**
The PUT issue endpoint will update the Maintenance Issue data with the request body data. 

### Checklist:

- [x] I have double checked that the issue does not currently exist. 
- [x] I have added the appropriate labels (e.g. feature, front-end, back-end).
- [x] I have added the **Platform Project** project

### PR DESCRIPTION
**Pull request by noahjade**
# Description

This page allows users to log maintenance issues. Note that it does Not send these to an external party, it simply allows flatmates to keep track of who has sent maintenance requests to the landlord or relevant parties. A use example could be the window is broken, Flatmate A checks Flatshare and sees that Flatmate B has left an entry documenting them contacting the landlord. So Flatmate A knows they don't have to contact the landlord.

This feature helps towards the goal of "Maintenance/Repairs - Documentation of complaints/issues and similar stuff" from the project proposal.

Fixes/resolves #141 

## Screenshots

Please include any screenshots or media that illustrates the changes made [to be added]

## Type of change

Please delete options that are not relevant.

- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

# Checklist:

- [x] Issue model created
- [x] Backend CRUD
- [X] Create Issue component (UI)
- [x] Unresolved issue component
- [x] Resolved issue component
- [ ] View issue detail page
- [x] Linking up backend and frontend
- [ ] Link to Issues page from the dashboard view

I have completed these steps when making this pull request:

- [X] I have assigned my name to the issue
- [x] I have moved the issue to the **In Progress** column
- [X] I have labelled the PR appropriately
- [X] I have assigned myself to the PR

Before opening the PR for review:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added attributions to new dependencies and resources
- [ ] I have moved the linked issue to the **Review in Progress** column

### CODE REVIEW

**Review by sher812 — DISMISSED** [review]
LGTM

File: packages/backend/src/controllers/issues/issues.controller.ts [hunk 1]
```diff
@@ -132,6 +133,37 @@ export class IssueController {
     );
   }
 
+  @Put(':id/resolve')
+  @ApiOperation({ summary: 'Mark a user as having reesolved an issue.' })
+  @ApiOkResponse({
+    description: 'Issue resolved successfully',
+    type: IssueResponseDto,
+  })
+  async updateBillPayment(
```
Comment (butterplup) [inline packages/backend/src/controllers/issues/issues.controller.ts]: can you change the name of `updateBillPayment` to `updateIssueResolved`  @sher812 and any other Bill/Issue discrepancies pls?

Comment (sher812) [reply packages/backend/src/controllers/issues/issues.controller.ts]: oh whoops

Comment (sher812) [reply packages/backend/src/controllers/issues/issues.controller.ts]: will do, do you also mean the variable names in the issue details page?

Comment (butterplup) [reply packages/backend/src/controllers/issues/issues.controller.ts]: yea, if you feel like it. not sure if anyone else has been assigned to do it yet. but it really helps my brain be less confused when reading these issues files 😅

**Pull request merged**